apps/web/src/api/v2/util.ts

```diff
@@ -9,12 +9,33 @@ import {
   Build,
   Crawl,
   Project,
+  PullRequest,
   ScreenshotBucket,
 } from "@argos-ci/database/models";
+import { getRedisLock } from "@argos-ci/web";
 
 export const getBuildName = (name: string | undefined | null) =>
   name || "default";
 
+const getOrCreatePullRequest = async ({
+  projectId,
+  number,
+}: {
+  projectId: string;
+  number: number;
+}) => {
+  const lockKey = `pullRequestCreation-${projectId}:${number}`;
```

Comment:
Remove projectId from PR, a PR is linked to a github repository if you want, but not to a project.